### DISCUSSION TITLE
How to insert text programmatically through JS  to CKEditor 5?

### DISCUSSION THREAD

**Discussion by SequelONE** [Laravel-Backpack/community-forum#634] [Q&A (Help)] [answered]
Hi all. I have a problem. 

When I click on the button, I get the generated ChatGPT text by ajax. I need to add this text with formatting. Generated text is indented `\n`

How to programmatically insert text via JS in **CKEditor 5**?

**Comment by pxpm** [comment, marked as answer]
Hey there @SequelONE 👋 

I asked our friend ChatGPT:
![image](https://github.com/Laravel-Backpack/community-forum/assets/7188159/39abfa9e-c44c-40b2-9b03-b3a55b43183f)

I guess is by adding a bit of Javascript to your page. 😃 

You can add a custom javascript file to your page with the code using [a script widget](https://backpackforlaravel.com/docs/6.x/base-widgets#script-1) 

Alternatively you can create a copy of `ckeditor` field `php artisan backpack:field ckeditor_custom --from=ckeditor` and register there some event listener that you send from the button with the generated text and you set the text on the editor. 

Don't forget to change the field in your controller to use your `ckeditor_custom` field.

Cheers
